docs/tutorials/built-your-own.md

```diff
@@ -0,0 +1,145 @@
+# Built your own skipper proxy
+
+One of the biggest advantages of skipper compared to other HTTP
+proxies is that skipper is a library first design. This means that it
+is common to built your custom proxy based on skipper.
+
+A minimal example project is
+[skipper-example-proxy](https://github.com/szuecs/skipper-example-proxy).
+
+```go
+/*
+This command provides an executable version of skipper with the default
+set of filters.
+
+For the list of command line options, run:
+
+    skipper -help
+
+For details about the usage and extensibility of skipper, please see the
+documentation of the root skipper package.
+
+To see which built-in filters are available, see the skipper/filters
+package documentation.
+*/
+package main
+
+import (
+	log "github.com/sirupsen/logrus"
+	lfilters "github.com/szuecs/skipper-example-proxy/filters"
+	"github.com/zalando/skipper"
+	"github.com/zalando/skipper/config"
+)
+
+func main() {
+	cfg := config.NewConfig()
+	if err := cfg.Parse(); err != nil {
+		log.Fatalf("Error processing config: %s", err)
+	}
+
+	log.SetLevel(cfg.ApplicationLogLevel)
+
+	opt := cfg.ToOptions()
+	opt.CustomFilters = append(opt.CustomFilters, lfilters.NewMyFilter())
+
+	log.Fatal(skipper.Run(opt))
+}
+```
+
+## Code
+Write the code and use the custom filter implemented in https://github.com/szuecs/skipper-example-proxy/blob/main/filters/custom.go
+```
+[:~]% mkdir -p /tmp/go/skipper
+[:~]% cd /tmp/go/skipper
+[:/tmp/go/skipper]% go mod init myproject
+go: creating new go.mod: module myproject
+[:/tmp/go/skipper]% cat >main.go
+package main
+
+import (
+	log "github.com/sirupsen/logrus"
+	lfilters "github.com/szuecs/skipper-example-proxy/filters"
+	"github.com/zalando/skipper"
+	"github.com/zalando/skipper/config"
+)
+
+func main() {
+	cfg := config.NewConfig()
+	if err := cfg.Parse(); err != nil {
+		log.Fatalf("Error processing config: %s", err)
+	}
+
+	log.SetLevel(cfg.ApplicationLogLevel)
+
+	opt := cfg.ToOptions()
+	opt.CustomFilters = append(opt.CustomFilters, lfilters.NewMyFilter())
+
+	log.Fatal(skipper.Run(opt))
+}
+CTRL-D
+[:/tmp/go/skipper]%
+```
+
+## Build
+I had some import failures because of a transitive dependency.
```

Comment:
Do we need to mention import failures here?

Reply:
ah sorry, yes good catch